### ISSUE TITLE
PM2 cluster for non http apps

### ISSUE THREAD

**Issue by alexstanciu** [Unitech/pm2#1739]
Hello,

How can I manage cluster messaging between master <--> workers? I'm not spawning a http server with node, I'm using it to process http requests from my servers to other web servers. I'm not opening any local port for my app. My app runs like manager -> worker(s) and I need a way to communicate between them. With node cluster you can process.send,worker.send. How is this working with pm2? I found: https://github.com/Unitech/PM2/blob/master/doc/PROGRAMMATIC.md but it looks really confusing to me.

**Comment by soyuka** [comment]
When using the embed `cluster` pm2 is the master. Here you want your app to be the master. So, you'll need to use child processes (`fork` or `cluster`) in your app and run it in `fork` mode.

**Comment by alexstanciu** [comment]
But like this in pm2 list I will see only one process running, even if i fork other workers, right?!
I'm planning to monitor also all spawns and pm2 show , list offers me good way of doing it. There is no way to control pm2 master process to list all my forks?

**Comment by soyuka** [comment]
Yes you could also use pm2 programmatically to fork many workers. Then, you'll be able to control them via pm2. I don't think that it'll expose an IPC channel (might be a nice feature though) but you could talk with your process through redis or [axon](https://www.npmjs.com/package/axon)!

**Comment by alexstanciu** [comment]
ok. Thanks a lot soyuka.

**Comment by soyuka** [comment]
Leave this open I want to investigate on the `IPC`, might be nice to programmatically get the `child_process` so that communication with workers gets easier.
